src/main/java/org/opensearch/neuralsearch/util/TokenWeightUtil.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class TokenWeightUtil {
+    public static String RESPONSE_KEY = "response";
+
+    /**
+     * possible input data format
+     * case remote inference:
+     * [{
+     *     "response":{
+     *         [
+     *         { TOKEN_WEIGHT_MAP},
+     *         { TOKEN_WEIGHT_MAP}
+     *         ]
+     *     }
+     * }]
+     * case local deploy:
+     * [{"response":{
+     *         [
+     *         { TOKEN_WEIGHT_MAP}
+     *         ]
+     *     }
+     * },{"response":{
+     *         [
+     *         { TOKEN_WEIGHT_MAP}
+     *         ]
+     *     }]
+     */
+    public static List<Map<String, Float>> fetchListOfTokenWeightMap(List<Map<String, ?>> mapResultList) {
+        List<Object> results = new ArrayList<>();
+        for (Map<String, ?> map : mapResultList) {
+            if (!map.containsKey(RESPONSE_KEY)) {
+                throw new IllegalArgumentException("The inference result should be associated with the field [" + RESPONSE_KEY + "].");
+            }
+            if (!List.class.isAssignableFrom(map.get(RESPONSE_KEY).getClass())) {
+                throw new IllegalArgumentException("The data object associated with field [" + RESPONSE_KEY + "] should be a list.");
+            }
+            results.addAll((List<?>) map.get("response"));
+        }
+        return results.stream().map(TokenWeightUtil::buildTokenWeightMap).collect(Collectors.toList());
+    }
+
+    private static Map<String, Float> buildTokenWeightMap(Object uncastedMap) {
+        if (!Map.class.isAssignableFrom(uncastedMap.getClass())) {
+            throw new IllegalArgumentException("The expected inference result is a Map with String keys and  " + " Float values.");
+        }
+        Map<String, Float> result = new HashMap<>();
+        for (Map.Entry<?, ?> entry : ((Map<?, ?>) uncastedMap).entrySet()) {
+            if (!String.class.isAssignableFrom(entry.getKey().getClass()) || !Number.class.isAssignableFrom(entry.getValue().getClass())) {
+                throw new IllegalArgumentException("The expected inference result is a Map with String keys and  " + " Float values.");
+            }
+            result.put((String) entry.getKey(), ((Number) entry.getValue()).floatValue());
```

Comment:
Is it possible getting a cast exception when the value is not a number?

Reply:
For remote inference the response is uncertain, it's possible to get a response with wrong data format. I suggest we throw Exception to help user find the problem.